astropy/coordinates/builtin_frames/astrometric.py

```diff
@@ -137,16 +136,57 @@ def icrs_to_astrometric(reference_frame, astrometric_frame):
         R = mat1 * mat2 * mat3
         return R
 
-    @frame_transform_graph.transform(DynamicMatrixTransform, Astrometric, framecls)
+    @frame_transform_graph.transform(DynamicMatrixTransform, _Astrometric, framecls)
     def astrometric_to_icrs(astrometric_coord, reference_frame):
-        """Convert an Astrometric frame coordinate to an ICRS"""
+        """Convert an Astrometric frame coordinate to the reference frame"""
 
         # use the forward transform, but just invert it
         R = icrs_to_astrometric(reference_frame, astrometric_coord)
         return R.T  # this is the inverse because R is a rotation matrix
 
-    _astrometric_cache[framecls] = Astrometric
-    return Astrometric
+    _astrometric_cache[framecls] = _Astrometric
+    return _Astrometric
+
+class AstrometricFrame(BaseCoordinateFrame):
+    """
+    A frame which is relative to some position on the sky. Useful for
+    calculating offsets and dithers in the frame of the sky.
```

Comment:
Can you add a reference from this `AstrometricFrame` docstring to the high-level docs section with explanations and examples on `AstrometricFrame`? Being able to go back and forth between high-level and API docs helps to grok what this is and how it works quickly.